src/main/java/com/vmware/vipclient/i18n/VIPCfg.java

```diff
@@ -71,56 +86,33 @@ public static synchronized VIPCfg getInstance() {
 		}
 		return gcInstance;
 	}
-	
-	/**
-	 * initialize the instance by parameter
-	 * 
-	 * @param vipServer
-	 * @param productName
-	 * @param version
-	 */
-	public void initialize(String vipServer, String productName, String version) {
-		this.productName = productName;
-		this.version = version;
-		this.vipServer = vipServer;
-	}
 
 	/**
 	 * initialize the instance by a properties file
-	 * 
+	 *
 	 * @param cfg
+	 * @throws IOException
+	 * @throws FileNotFoundException
 	 */
-	public void initialize(String cfg) {
-		ResourceBundle prop = ResourceBundle.getBundle(cfg);
-		if (prop == null) {
-			return; 
+	@SuppressWarnings("serial")
+	public void initialize(String cfg) throws IOException {
+		InputStream stream = ClassLoader.getSystemResourceAsStream(cfg);
+		LinkedHashMap<String, Object> data = new Yaml().loadAs(stream, (new LinkedHashMap<String, Object>() {}).getClass());
+    	
+		for (Entry<String, Object> entry : data.entrySet()) {
+			try {
+				this.getClass().getDeclaredField(entry.getKey()).set(this, entry.getValue());
+			} catch (IllegalArgumentException e) {
+				throw new VIPJavaClientException(
+						String.format("Invalid value '%s' for setting '%s'!", entry.getValue(), entry.getKey()), e);
+			} catch (NoSuchFieldException e) {
+				throw new VIPJavaClientException("Invalid setting item: " + entry.getKey());
+			} catch (SecurityException | IllegalAccessException e) {
+				throw new VIPJavaClientException("Unknow errorr");
+			}
 		}
 		
-		if (prop.containsKey("productName"))
-			this.productName = prop.getString("productName");
-		if (prop.containsKey("version"))
-			this.version = prop.getString("version");
-		if (prop.containsKey("vipServer"))
-			this.vipServer = prop.getString("vipServer");
-		if (prop.containsKey("pseudo"))
-			this.pseudo = Boolean.parseBoolean(prop.getString("pseudo"));
-		if (prop.containsKey("collectSource"))
-			this.collectSource = Boolean.parseBoolean(prop
-					.getString("collectSource"));
-		if (prop.containsKey("initializeCache"))
-			this.initializeCache = Boolean.parseBoolean(prop
-					.getString("initializeCache"));
-		if (prop.containsKey("cleanCache"))
-			this.cleanCache = Boolean.parseBoolean(prop
-					.getString("cleanCache"));
-		if (prop.containsKey("machineTranslation"))
-			this.machineTranslation = Boolean.parseBoolean(prop
-					.getString("machineTranslation"));
-		if (prop.containsKey("i18nScope"))
-			this.i18nScope = prop.getString("i18nScope");
-		if (prop.containsKey("cacheExpiredTime"))
-			this.cacheExpiredTime = Long.parseLong(prop
-					.getString("cacheExpiredTime"));
+		StringService.loadResources(this.resources);
```

Comment:
The VIPCfg class should provide a method getResources for the instance of StringService to call, so 'StringService.loadResources(this.resources)' should be removed?